src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -792,20 +820,24 @@ public static void ActionMethodExecuting(this ILogger logger, ControllerContext
                     {
                         convertedArguments[i] = Convert.ToString(arguments[i]);
                     }
-                }
 
-                var validationState = context.ModelState.ValidationState;
-
-                _actionMethodExecuting(logger, actionName, convertedArguments, validationState, null);
+                    _actionMethodExecutingWithArguments(logger, actionName, convertedArguments, validationState, null);
+                }
             }
         }
 
-        public static void ActionMethodExecuted(this ILogger logger, ControllerContext context, IActionResult result)
+        public static void ActionMethodExecuted(this ILogger logger, ControllerContext context, IActionResult result, long startTimestamp)
         {
-            if (logger.IsEnabled(LogLevel.Debug))
+            if (logger.IsEnabled(LogLevel.Information))
             {
-                var actionName = context.ActionDescriptor.DisplayName;
-                _actionMethodExecuted(logger, actionName, Convert.ToString(result), null);
+                if (startTimestamp != 0)
+                {
+                    var currentTimestamp = Stopwatch.GetTimestamp();
+                    var elapsed = new TimeSpan((long)(TimestampToTicks * (currentTimestamp - startTimestamp)));
```

Comment:
This is one of the nice things about valuestopwatch, it does all of this! https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.ValueStopwatch.Sources/ValueStopwatch.cs